src/ludamus/gates/web/django/templatetags/markdown_tags.py

```
@@ -1,13 +1,25 @@
+    def _sub(m: Match[str]) -> str:
+        slash, level, attrs = m.group(1), int(m.group(2)), m.group(3) or ""
+        return f"<{slash}h{min(level + 2, 6)}{'' if slash else attrs}>"
+    return _HEADING_RE.sub(_sub, html)
+
 
 @register.filter
 def render_markdown(text: str) -> str:
     if not text:
         return ""
-    return mark_safe(_render_markdown(text))  # noqa: S308
+    return mark_safe(_shift_headings(_render_markdown(text)))
```

Ruff (0.15.18)

[error] Use of `mark_safe` may expose cross-site scripting vulnerabilities

(S308)